public/app/core/components/IntervalInput/IntervalInput.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+
+import { InlineField, InlineFieldRow, Input } from '@grafana/ui';
+
+import { validateInterval } from './validation';
+
+interface Props {
+  label: string;
+  tooltip: string;
+  value: string;
+  onChange: (val: string) => void;
+  isInvalidError: string;
+  disabled?: boolean;
+}
+
+export function IntervalInput(props: Props) {
+  const [interval, setInterval] = useState(props.value);
```

Comment:
The debouncing now makes more sense to me but now it is used only for the validation. So do we need to have the local state? Does not seem so, we update parent state via `onChange` and we should get new `value` right away. In that sense having the local value does not seem to be needed. Actually probably creates a possible bug because if the value changes from outside, that update is not reflected in the local `interval` and we would not reflect the `prop.value` in any way.

Reply:
ahh, yes.